### ISSUE TITLE
Render text with ligatures

### ISSUE THREAD

**Issue by EngRajabi** [SixLabors/ImageSharp.Drawing#81]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp.Drawing
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp.Drawing/issues) and [closed](https://github.com/SixLabors/ImageSharp.Drawing/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Write text in an image when the text is utf8. Broken is shown.
sample: 
![Untitled](https://user-images.githubusercontent.com/10164223/90049133-a4e23400-dce9-11ea-9a73-a3fb00b2dd96.jpg)

### System Configuration

- ImageSharp.Drawing version: latest 
- Other ImageSharp packages and versions: latest 
- Environment (Operating system, version and so on): windows 10
- .NET Framework version: 3.1
- Additional information:

**Comment by tocsoft** [comment]
What version do you mean as 'latest'... the word 'latest' doesn't age well if we still have this issue in 6months after we have release multiple new releases then we need to know when it was first seen etc.

Can you please provide details about what font your using if possible attach the actual font file that cause issues..
Can you please provide a code sample that we can run to reproduce the issue.
Can you describe/include a picture of what the expected results are... none of the team read/write that language and have no idea what would be wrong with it.

**Comment by EngRajabi** [comment]
drawing version: 1.0.0-beta0010
image sharp version: 1.0.0
font : B Nazanin.ttf

code: SixLabors.ImageSharp.Drawing.Tests.Drawing.Text.DrawTextOnImageTests
```csharp

        private const string AB = "رجبی";

        private const string TestText = "محسن رجبی";

        [Theory]
        [WithSolidFilledImages(200, 150, "White", PixelTypes.Rgba32, 50, 0, 0, "B Nazanin.ttf", AB)]
        [WithSolidFilledImages(900, 150, "White", PixelTypes.Rgba32, 50, 0, 0, "B Nazanin.ttf", TestText)]
        [WithSolidFilledImages(1100, 200, "White", PixelTypes.Rgba32, 50, 150, 50, "B Nazanin.ttf", TestText)]
        public void FontShapesAreRenderedCorrectlyWithAPen<TPixel>(
            TestImageProvider<TPixel> provider,
            int fontSize,
            int x,
            int y,
            string fontName,
            string text)
            where TPixel : unmanaged, IPixel<TPixel>
        {
            Font font = CreateFont(fontName, 30);
            var color = Color.Black;

            provider.VerifyOperation(
                OutlinedTextDrawingComparer,
                img =>
                {
                    img.Mutate(c => c.DrawText(text, new Font(font, fontSize), null, Pens.Solid(color, 1), new PointF(x, y)));

                    img.Save(@"D:\img.jpg");
                },
                $"pen_{fontName}-{fontSize}-{ToTestOutputDisplayText(text)}-({x},{y})",
                appendPixelTypeToFileName: false,
                appendSourceFileOrDescription: true);
        }
```

**Comment by tocsoft** [comment]
awesome... thanks for that.

ok is the problem then that the letters are outlined rather than filled in?

**Comment by EngRajabi** [comment]
The right word: "محسن رجبی"
The word printed on the photo: "ی ب ج ر  ن س ح م"

**Comment by tocsoft** [comment]
ah ok. That should be enough so that someone can start investigating.

**Comment by tocsoft** [comment]
In the mean time something you should be aware of is we don't actually have any support real for true right to left rendering... closest we have is we can right align left to right text.

Same actually applies to top to bottom.

**Comment by tocsoft** [comment]
For any future people (probably myself) looks like this probably actually due the the lack of ligature support in https://github.com/SixLabors/Fonts

https://github.com/SixLabors/Fonts/issues/150 has been raised to track this.

Might need follow up work for wider Right to Left support to ensure word ordering and punctuation placement is correct.

**Comment by EngRajabi** [comment]
update

**Comment by JimBobSquarePants** [comment]
>update

Please don't leave comments like this. We are under no obligation to update this for you and will do so in our own time. If you wish to expediate the process please submit a Pull Request with an appropriate fix.

**Comment by EngRajabi** [comment]
First of all, thank you for developing this framework, but you, as a professional, should be able to use it for everyone when designing such a package.  At present, 1.8 billion people use rtl languages.

https://www.worldatlas.com/amp/articles/which-languages-are-written-from-right-to-left.html

**Comment by JimBobSquarePants** [comment]
It’s an open source project released under a license with no guarantees. We’re under no obligation to write it for anybody. 

Writing software takes time. As yet none of these 1.8 billion people have offered up their time to help develop the feature.

**Comment by huysentruitw** [comment]
@EngRajabi, 

Professionals like to get payed. If you really want this feature but don't want to create a PR yourself, have you considered sponsoring?

**Comment by hishamco** [comment]
@JimBobSquarePants I may have look to this one because it's my pleasure to support RTL text in image

**Comment by VahidN** [comment]
lib-gdi-plus [uses Pango](https://github.com/mono/libgdiplus/search?q=Pango&type=code) library to resolve this issue:
https://pango.gnome.org/

**Comment by JimBobSquarePants** [comment]
Thanks @VahidN but one of the targets of this library is to build everything in managed C#

**Comment by antonfirsov** [comment]
Looks like duplicate of #17, closing to clean up the backlog. Let's reopen if I'm missing something.

**Comment by JimBobSquarePants** [comment]
I've just pushed an update into the MyGet feed `1.0.0-beta13.15` which adds Bidi support to text drawing including ligature shaping. 

Arabic and Latin-based shaping is complete Please bear in mind that there's are still some [language specific font shapers to implement](https://github.com/SixLabors/Fonts/issues?q=is%3Aissue+is%3Aopen+label%3Ashaping) (_help needed and appreciated there_) so there might still be some quirks.

**Comment by VahidN** [comment]
@JimBobSquarePants
Thank you. Would you please upload it to NuGet as well?

**Comment by JimBobSquarePants** [comment]
>Thank you. Would you please upload it to NuGet as well?

@VahidN Not until the library has reached RC1 quality. I'd like to get some of these shapers implemented first but I need help.

https://github.com/SixLabors/ImageSharp.Drawing/issues?q=is%3Aissue+is%3Aopen+label%3Aarea%3Ashapes
